pkg/fleet-manager/application_helper.go

```diff
@@ -0,0 +1,490 @@
+	currentResource.Spec = targetSource.Spec
+	if err := a.Client.Update(ctx, currentResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+// updateHelmRepository updates the state of a current HelmRepository resource to match the provided target HelmRepository resource.
+// This function is used by syncResource to keep the actual state of HelmRepository resources in sync with the desired state.
+func (a *ApplicationManager) updateHelmRepository(ctx context.Context, currentResource *sourceapi.HelmRepository, targetSource *sourceapi.HelmRepository) error {
+	currentResource.Spec = targetSource.Spec
+	if err := a.Client.Update(ctx, currentResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+// updateOCIRepository updates the state of a current OCIRepository resource to match the provided target OCIRepository resource.
+// This function is used by syncResource to keep the actual state of OCIRepository resources in sync with the desired state.
+func (a *ApplicationManager) updateOCIRepository(ctx context.Context, currentResource *sourceapi.OCIRepository, targetSource *sourceapi.OCIRepository) error {
+	currentResource.Spec = targetSource.Spec
+	if err := a.Client.Update(ctx, currentResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+// updateKustomization updates the state of a current Kustomization resource to match the provided target Kustomization resource.
+// This function is used by syncResource to keep the actual state of Kustomization resources in sync with the desired state.
+func (a *ApplicationManager) updateKustomization(ctx context.Context, currentResource *kustomizev1.Kustomization, targetSource *kustomizev1.Kustomization) error {
+	currentResource.Spec = targetSource.Spec
+	if err := a.Client.Update(ctx, currentResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+// updateHelmRelease updates the state of a current HelmRelease resource to match the provided target HelmRelease resource.
+// This function is used by syncResource to keep the actual state of HelmRelease resources in sync with the desired state.
+func (a *ApplicationManager) updateHelmRelease(ctx context.Context, currentResource *helmv2beta1.HelmRelease, targetSource *helmv2beta1.HelmRelease) error {
+	currentResource.Spec = targetSource.Spec
+	if err := a.Client.Update(ctx, currentResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+// findSourceKind determines the type of the application's source.
+// An application can only have one specified source type. In case of none or multiple source types are specified,
+// this function returns an error. If successful, it returns the string representation of the source type.
+func findSourceKind(app *applicationapi.Application) (string, error) {
+	validCount := 0
+	validKind := ""
+
+	// Check for each type of source in the application and count the number of valid sources
+	gitRepo := app.Spec.Source.GitRepo
```

Comment:
you should not do these check here, it should be done in validating webhook.

Reply:
add TODO